Tests/LibWasm/test-wasm.cpp

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2021, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/File.h>
+#include <LibTest/JavaScriptTestRunner.h>
+#include <LibWasm/AbstractMachine/Interpreter.h>
+#include <LibWasm/Types.h>
+
+TEST_ROOT("Userland/Libraries/LibWasm/Tests");
+
+TESTJS_GLOBAL_FUNCTION(read_binary_wasm_file, readBinaryWasmFile)
+{
+    auto filename = vm.argument(0).to_string(global_object);
+    if (vm.exception())
+        return {};
+    auto file = Core::File::open(filename, Core::OpenMode::ReadOnly);
+    if (file.is_error()) {
+        vm.throw_exception<JS::TypeError>(global_object, file.error());
+        return {};
+    }
+    auto contents = file.value()->read_all();
+    auto array = JS::Uint8Array::create(global_object, contents.size());
+    contents.span().copy_to(array->data());
+    return array;
+}
+
+class WebAssemblyModule final : public JS::Object {
+    JS_OBJECT(WebAssemblyModule, JS::Object);
+
+public:
+    explicit WebAssemblyModule(JS::Object& prototype)
+        : JS::Object(prototype)
+    {
+    }
+
+    static Wasm::AbstractMachine& machine() { return m_machine; }
+    Wasm::Module& module() { return *m_module; }
+    Wasm::ModuleInstance& module_instance() { return *m_module_instance; }
+
+    static WebAssemblyModule* create(JS::GlobalObject& global_object, Wasm::Module module)
+    {
+        auto instance = global_object.heap().allocate<WebAssemblyModule>(global_object, *global_object.object_prototype());
+        instance->m_module = move(module);
+        if (auto result = machine().instantiate(*instance->m_module, {}); result.is_error())
+            global_object.vm().throw_exception<JS::TypeError>(global_object, result.release_error().error);
+        else
+            instance->m_module_instance = result.release_value();
+        return instance;
+    }
+    void initialize(JS::GlobalObject&) override;
+
+    ~WebAssemblyModule() override = default;
+
+private:
+    JS_DECLARE_NATIVE_FUNCTION(get_export);
+    JS_DECLARE_NATIVE_FUNCTION(wasm_invoke);
+
+    static Wasm::AbstractMachine m_machine;
+    Optional<Wasm::Module> m_module;
+    OwnPtr<Wasm::ModuleInstance> m_module_instance;
+};
+
+Wasm::AbstractMachine WebAssemblyModule::m_machine;
+
+TESTJS_GLOBAL_FUNCTION(parse_webassembly_module, parseWebAssemblyModule)
+{
+    auto object = vm.argument(0).to_object(global_object);
```

Comment:
```suggestion
    auto object = vm.argument(0).to_object(global_object);
    if (vm.exception())
        return {};
```

Reply:
Ah, these were in the previous PR...sorry I missed them

Reply:
by "PR" I mean commit...